/*
    Написать функцию, определяющую, является ли счастливым 6-значное число.
    Функция должна выбрасывать два исключения.
    Первое исключение - если число содержит 5 или менее цифр.
    Второе исключение - если число содержит 7 или более цифр.
    Второе исключение выбросите с использованием класса exception.
    Исключения обрабатываем в main()
*/
#include <iostream>
using namespace std;

void Check(int numb)
{
   int i = 0;       // управляющая переменная цикла
   int n = numb;    // копия введенного числа

    for (; n > 0; i++)  // цикл для проверки ко-ва цифр в числе
    {
        n /= 10;
    }
    if (i <= 5)         // если кол-во цифр меньше 6, то бросаем исключение
    {
        throw "\n\n Ошибка! Число содержит меньше 6 цифр\n\n";
    }
    if (i >= 7)         // если кол-во цифр больше 6, то бросаем исключение exception
    {
        throw exception("\n\n Ошибка! Число содержит больше 6 цифр!\n\n");
    }

    short n6 = numb % 10;          // разбиваем число на цифры
    short n5 = numb / 10 % 10;
    short n4 = numb / 100 % 10;
    short n3 = numb / 1000 % 10;
    short n2 = numb / 10000 % 10;
    short n1 = numb / 100000;

    short sum1 = n1 + n2 + n3;    // находим сумму первых трех цифр
    short sum2 = n4 + n5 + n6;    // находим сумму вторых трех цифр

    // если суммы равны, то билет счастливый, иначе - не счастливый
    cout << ((sum1 == sum2) ? " Ваш билет является счастливым!\n" : " Ваш билет не является счастливым!\n");
}
int main()
{
    setlocale(0, "rus");
    cout << " Счастливый билет \n\n";

    cout << " Введите 6-значный номер билета: \n";
    int numb; 
    cin >> numb;

    try
    {
        Check(numb);
    }
    catch (const char* s) 
    {
        cout << s;
    }
    catch (const exception& ex)
    {
        cout << ex.what();
    }
    catch (...)
    {
        cout << "\n\n Вы ввели неверное число!\n\n";
    }
    return 0;
}